### PR TITLE
ENH: Improve KLCompareHistogramImageToImageMetric coverage

### PR DESCRIPTION
**Pull request by jhlegarreta**
Increase `itk::KullbackLeiblerCompareHistogramImageToImageMetric` coverage:
- Exercise the basic object methods using the
  `ITK_EXERCISE_BASIC_OBJECT_METHODS` macro.
- Test the Set/Get methods using the `ITK_TEST_SET_GET_VALUE` macro.

Take advantage of the commit to improve the style:
- Only enclose within the `ITK_TRY_EXPECT_NO_EXCEPTION` macro the filter
  update calls that may throw exceptions.
- Add braces to control flow blocks even if composed with a single
  statement to comply with the ITK style.
- Mark the test ending according to the ITK style.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Should make the header file reach 100% coverage:
https://open.cdash.org/viewCoverageFile.php?buildid=7490974&fileid=48288255

I did not want to change the epsilon parameter value or introduce it as an argument to the test as it has no other arguments, but if anybody feels I should, I will do it.

**Comment by jhlegarreta** [comment]
Tests [are passing on CircleCI](https://app.circleci.com/pipelines/github/InsightSoftwareConsortium/ITK/3482/workflows/fe23ca0c-3c44-43d2-8725-8d0d5c3c1915/jobs/10239?invite=true#step-109-3795):
```
100% tests passed, 0 tests failed out of 2884
```

It is just that it was [unable to submit the report to CDash](https://app.circleci.com/pipelines/github/InsightSoftwareConsortium/ITK/3482/workflows/fe23ca0c-3c44-43d2-8725-8d0d5c3c1915/jobs/10239?invite=true#step-109-3912):
```
Submit files
   SubmitURL: https://open.cdash.org/submit.php?project=Insight
   Error when uploading file: /home/circleci/ITK-build/Testing/20211002-2349/Update.xml
   Error message was: SSL certificate problem: certificate has expired
   Problems when submitting via HTTP
```

**Review by thewtex — APPROVED** [review]
@jhlegarreta awesome, thanks!!

**Comment by thewtex** [comment]
CircleCI unrelated re #2765

File: Modules/Registration/Common/test/itkCompareHistogramImageToImageMetricTest.cxx [hunk 1]
```diff
@@ -20,138 +20,139 @@
...
-    fixedImageSource->Update();  // Force the filter to run
+  fixedImageSource->SetSize(fixedImageSize);
+  fixedImageSource->SetOrigin(fixedImageOrigin);
+  fixedImageSource->SetSpacing(fixedImageSpacing);
+  fixedImageSource->SetNormalized(false);
+  fixedImageSource->SetScale(250.0f);
 
-    MovingImageType::Pointer movingImage = movingImageSource->GetOutput();
-    FixedImageType::Pointer  fixedImage = fixedImageSource->GetOutput();
+  ITK_TRY_EXPECT_NO_EXCEPTION(movingImageSource->Update()); // Force the filter to run
```
Comment (dzenanz) [inline Modules/Registration/Common/test/itkCompareHistogramImageToImageMetricTest.cxx:70]: Can this fail? I guess it can fail to allocate memory.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**